lib/Doctrine/DBAL/Statement.php

```diff
@@ -155,6 +158,12 @@ public function bindParam($param, &$variable, $type = ParameterType::STRING, $le
      */
     public function execute($params = null)
     {
+        Deprecation::triggerIfCalledFromOutside(
+            'doctrine/dbal',
+            'https://github.com/doctrine/dbal/pull/4580',
```

Comment:
#4569 only reports this as a bug, because ::execute still returns a boolean. This PR here has all the summary about why this change makes sense and examples of new/old, I would prefer this as a user trying to understand the deprecation. Should I open a new issue that copies over the summary?

Reply:
No, let's keep it simple. But let's give it some thought in the meanwhile. Technically, both the PR and the issue are discoverable via Git history. But as an end-user, I might be more interested in "why" it was deprecated (what the problem with this API is, the issue) rather than "how it was deprecated" (the code change). Just thinking out loud.